library/std/src/thread/mod.rs

```diff
@@ -1293,6 +1293,23 @@ impl<'scope, T> Drop for Packet<'scope, T> {
             // panicked, and nobody consumed the panic payload, we make sure
             // the scope function will panic.
             let unhandled_panic = matches!(self.result.get_mut(), Some(Err(_)));
+            // Drop the result before decrementing the number of running
+            // threads, because the Drop implementation might still use things
+            // it borrowed from 'scope.
+            // This is only relevant for threads that aren't join()ed, as
+            // join() will take the `result` and set it to None, such that
+            // there is nothing left to drop here.
+            // If this drop panics, that just results in an abort, because
```

Comment:
Reading just this comment in isolation, I feel like it is dangerous to assume that all unwinding implementations would abort in absence of landing pads on the stack. I suspect there won't be any documented mandate for this sort of behavior anywhere.

Reply:
Maybe, but then that's a larger problem that Rust's regular threads (through std::thread::spawn) also have (and had, since like forever).

Reply:
It's easy to fix it for both cases by adding a catch_unwind and abort here though. I'll do that in a second commit.

Reply:
Added a second commit that fixes that issue for both scoped and regular threads. It now explicitly calls `rtabort!()` when dropping the thread result panics.

Reply:
(cc https://github.com/rust-lang/lang-team/issues/97, which would make this a non-issue.)